app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,117 @@
+class ReviewsController < ApplicationController
+
+  authorize_resource
+
+  def index
+    @reviews = Review.all
+    @reviews_page = @reviews.page(params[:page])
+    create_report
+  end
+
+  def search_results
+    #fail params.inspect
+    if params[:search_keywords].empty?
+      @reviews_found = Review.date_search(params)
+    elsif params[:start_date].empty? and params[:end_date].empty?
+      @reviews_found = Review.university_search(params)
+      if @reviews_found.blank?
+        @reviews_found = Review.keyword_search(params)
+      end
+    else
+      @reviews_found = Review.keyword_search(params).date_search(params)
+    end
+    @reviews_page = @reviews_found # чтобы не создавать новый row
+  end
+
+  def show
+    @review = Review.find(params[:id])
+  end
+
+  def new
+    @review = Review.new
+    @university = University.all
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+    count_total
+  end
+
+  def create
+    @review = Review.new(resource_params)
+    count_total
+    if @review.save
+      redirect_to @review
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    count_total
+    if @review.update(resource_params)
+      redirect_to @review
+    else
+      render 'edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.